relay-general/src/protocol/security_report.rs

```diff
@@ -216,22 +216,34 @@ impl CspRaw {
         "self".to_string()
     }
 
-    fn effective_directive(&self) -> Result<CspDirective, InvalidSecurityError> {
+    fn effective_directive(&self) -> Result<(CspDirective, String), InvalidSecurityError> {
```

Comment:
possible alternative: return an Option<String> with the Some-variant only when the longer format is present